axum-core/src/extract/request_parts.rs

```diff
@@ -79,14 +111,47 @@ where
     type Rejection = BytesRejection;
 
     async fn from_request(req: Request, _: &S) -> Result<Self, Self::Rejection> {
-        let bytes = req
-            .into_limited_body()
-            .collect()
+        let mut body = req.into_limited_body();
+
+        // If there's only 1 chunk, we can just return Buf::to_bytes()
```

Comment:
Nit, we're not returning `Buf::to_bytes()` though. Maybe something like

> If there's only 1 chunk, we can just return it directly

?

Feel free to ignore though.

Reply:
```suggestion
        // If there's only 1 chunk, we can just return it as-is
```